src/core/compact_object.cc

```diff
@@ -556,6 +556,20 @@ void RobjWrapper::SetString(string_view s, MemoryResource* mr) {
   }
 }
 
+void RobjWrapper::ReserveString(size_t size, MemoryResource* mr) {
+  type_ = OBJ_STRING;
+  encoding_ = OBJ_ENCODING_RAW;
+  MakeInnerRoom(0, size, mr);
+  sz_ = 0;
```

Comment:
why `sz_ = 0;` ?
MakeInnerRoom - copies the string if it existed before.

Reply:
now I see the usecase. Maybe just check that `inner_obj() == nullptr` at line 560?